test/test_jit.py

```diff
@@ -11188,6 +11188,21 @@ def return_tuple(x):
             return a, x
         self.checkScript(return_tuple, (torch.rand(4),))
 
+    def test_add_tuple_slice(self):
+        def foo(input: Tuple[torch.Tensor, Optional[torch.Tensor], Optional[torch.Tensor]]):
+            changed_input = input[0] + 1
+            return (changed_input,) + input[1:]
+        inp: Tuple[torch.Tensor, Optional[torch.Tensor], Optional[torch.Tensor]] = (torch.rand(4), None, None)
+        self.checkScript(foo, (inp,))
```

Comment:
Yay! This will make the users of the hooks much happier until we can get the type inferred hooks PR written